packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -2126,7 +2108,66 @@ abstract class TreeDiagnosticsMixin implements TreeDiagnostics {
   ///
   /// Used by [toStringDeep], [toDiagnosticsNode] and [toStringShallow].
   @protected
-  List<DiagnosticsNode> debugDescribeChildren() {
-    return const <DiagnosticsNode>[];
+  List<DiagnosticsNode> debugDescribeChildren() => const <DiagnosticsNode>[];
+}
+
+/// A class that can be used as a mixin that helps dump string and
+/// [DiagnosticsNode] representations of trees.
+///
+/// This class is identical to DiagnosticableTree except that it can be used as
+/// a mixin.
+abstract class DiagnosticableTreeMixin implements DiagnosticableTree {
+  // This class is intended to be used as a mixin, and should not be
+  // extended directly.
+  factory DiagnosticableTreeMixin._() => null;
+
+  @override
+  String toString() {
+    return toDiagnosticsNode(style: DiagnosticsTreeStyle.singleLine).toString();
+  }
+
+  @override
+  String toStringShallow([String joiner = '; ']) {
+    final StringBuffer result = new StringBuffer();
+    result.write(toShortDescription());
+    result.write(joiner);
+    final List<DiagnosticsNode> properties = <DiagnosticsNode>[];
+    debugFillProperties(properties);
+    result.write(
+      properties.where((DiagnosticsNode n) => !n.hidden).join(joiner),
+    );
+    return result.toString();
   }
+
+  @override
+  String toStringDeep([String prefixLineOne = '', String prefixOtherLines = '']) {
+    return toDiagnosticsNode().toStringDeep(prefixLineOne, prefixOtherLines);
+  }
+
+  @override
+  String toShortDescription() => describeIdentity(this);
+
+  @override
+  DiagnosticsNode toDiagnosticsNode({ String name, DiagnosticsTreeStyle style }) {
+    return new _DiagnosticableTreeNode(
+      name: name,
+      value: this,
+      style: style ?? debugDefaultDiagnosticsTreeStyle,
+    );
+  }
+
+  @override DiagnosticsTreeStyle get debugDefaultDiagnosticsTreeStyle {
+    return DiagnosticsTreeStyle.sparse;
+  }
+
+  @override String get debugEmptyBodyDescription => null;
+
+  @override
+  @protected
```

Comment:
having both override and protected seems like an error. Only the place that introduces the member should say protected. If you have override, you shouldn't have anything else.